File: client-clojure/src/sirqul_io_t_platform/specs/list_count_response.clj
(ns sirqul-io-t-platform.specs.list-count-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            )
  (:import (java.io File)))


(def list-count-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :locationOffersCount) int?
   (ds/opt :instantOffersCount) int?
   (ds/opt :allOffersCount) int?
   (ds/opt :returning) string?
   })

(def list-count-response-spec
  (ds/spec
    {:name ::list-count-response
     :spec list-count-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/ListCountResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ListCountResponse._

case class ListCountResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
locationOffersCount: Option[Long],
instantOffersCount: Option[Long],
allOffersCount: Option[Long],
returning: Option[String])

object ListCountResponse {
  import DateTimeCodecs._

  implicit val ListCountResponseCodecJson: CodecJson[ListCountResponse] = CodecJson.derive[ListCountResponse]
  implicit val ListCountResponseDecoder: EntityDecoder[ListCountResponse] = jsonOf[ListCountResponse]
  implicit val ListCountResponseEncoder: EntityEncoder[ListCountResponse] = jsonEncoderOf[ListCountResponse]
}
